pkg/manifests/manifests.go

```diff
@@ -1441,6 +1441,10 @@ func (f *Factory) PrometheusUserWorkload(grpcTLS *v1.Secret) (*monv1.Prometheus,
 		p.Spec.RemoteWrite = f.config.UserWorkloadConfiguration.Prometheus.RemoteWrite
 	}
 
+	if f.config.UserWorkloadConfiguration.Prometheus.EnforcedSampleLimit != nil {
```

Comment:
This is a new configuration field so we do not want to add this to the old ConfigMap as a way to encourage users to migrate away from the old configmap as it will be deprecated in 4.7.

Reply:
precisely why having separate structs for configs is a really good idea :+1: :100: 

very nice!